addons/controllers/tanzuclusterbootstrap_controller.go

```diff
@@ -584,3 +598,77 @@ func (r *TanzuClusterBootstrapReconciler) watchProvider(providerRef *corev1.Type
 		},
 	)
 }
+
+func (r *TanzuClusterBootstrapReconciler) reconcileCNISelection(
```

Comment:
I would suggest something below that gets called after `tanzuClusterBootstrap.Spec = tanzuClusterBootstrapTemplate.Spec.DeepCopy()` instead of patching the object after it is created.
```
func (r *TanzuClusterBootstrapReconciler) getCNI(tanzuClusterBootstrapTemplate *runtanzuv1alpha3.TanzuClusterBootstrapTemplate, cluster *clusterapiv1beta1.Cluster, log logr.Logger) (*runtanzuv1alpha3.TanzuClusterBootstrapPackage, error)
```
Once CNIs has been set and tanzuClusterBootstrap object created CNIs is immutable and will be enforced by a webhook.